bleak/_api.py

```diff
@@ -0,0 +1,237 @@
+        BleakGATTDescriptorBlueZDBus as _BleakGATTDescriptorImplementation,
+    )  # noqa: F401
+elif platform.system() == "Darwin":
+    try:
+        from CoreBluetooth import CBPeripheral  # noqa: F401
+    except Exception as ex:
+        raise BleakError("Bleak requires the CoreBluetooth Framework") from ex
+
+    from bleak.backends.corebluetooth.scanner import (
+        BleakScannerCoreBluetooth as _BleakScannerImplementation,
+    )  # noqa: F401
+    from bleak.backends.corebluetooth.device import (
+        BLEDeviceCoreBluetooth as _BLEDeviceImplementation,
+    )  # noqa: F401
+    from bleak.backends.corebluetooth.client import (
+        BleakClientCoreBluetooth as _BleakClientImplementation,
+    )  # noqa: F401
+    from bleak.backends.corebluetooth.service import (
+        BleakGATTServiceCoreBluetooth as _BleakGATTServiceImplementation,
+    )  # noqa: F401
+    from bleak.backends.corebluetooth.characteristic import (
+        BleakGATTCharacteristicCoreBluetooth as _BleakGATTCharacteristicImplementation,
+    )  # noqa: F401
+    from bleak.backends.corebluetooth.descriptor import (
+        BleakGATTDescriptorCoreBluetooth as _BleakGATTDescriptorImplementation,
+    )  # noqa: F401
+elif platform.system() == "Windows":
+    # Requires Windows 10 Creators update at least, i.e. Window 10.0.16299
+    _vtup = platform.win32_ver()[1].split(".")
+    if int(_vtup[0]) != 10:
+        raise BleakError(
+            "Only Windows 10 is supported. Detected was {0}".format(
+                platform.win32_ver()
+            )
+        )
+
+    if (int(_vtup[1]) == 0) and (int(_vtup[2]) < 16299):
+        raise BleakError(
+            "Requires at least Windows 10 version 0.16299 (Fall Creators Update)."
+        )
+
+    from bleak.backends.winrt.scanner import (
+        BleakScannerWinRT as _BleakScannerImplementation,
+    )  # noqa: F401
+    from bleak.backends.device import (
+        BLEDevice as _BLEDeviceImplementation,
+    )  # noqa: F401
+    from bleak.backends.winrt.client import (
+        BleakClientWinRT as _BleakClientImplementation,
+    )  # noqa: F401
+    from bleak.backends.winrt.service import (
+        BleakGATTServiceWinRT as _BleakGATTServiceImplementation,
+    )  # noqa: F401
+    from bleak.backends.winrt.characteristic import (
+        BleakGATTCharacteristicWinRT as _BleakGATTCharacteristicImplementation,
+    )  # noqa: F401
+    from bleak.backends.winrt.descriptor import (
+        BleakGATTDescriptorWinRT as _BleakGATTDescriptorImplementation,
+    )  # noqa: F401
+else:
+    raise BleakError(f"Unsupported platform: {platform.system()}")
+
+# Now let's tie together the abstract class and the backend implementation
+class BleakScanner(_BleakScannerImplementation):
+    """
+    Interface for Bleak Bluetooth LE Scanners.
+
+    The actual implementation is dependent on the backend used, and some methods (notably the
+    constructor) may have additional optional arguments.
```

Comment:
For me, one of the main reasons for making this change is to be able to have all of the OS-specific quirks/args listed in one place rather than having read the same docs 5 times for each platform. This makes it seem like we are not accomplishing that goal.

Reply:
Yes, this needs fixing.

Reply:
I think this is fixed with the `abstract_api` changes.